src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -649,7 +649,7 @@ private boolean checkFilteredAliases(Resolved requestedResolved, String action,
             indexMetaDataCollection = new Iterable<IndexMetadata>() {
                 @Override
                 public Iterator<IndexMetadata> iterator() {
-                    return clusterService.state().getMetadata().getIndices().valuesIt();
+                    return (Iterator<IndexMetadata>) clusterService.state().getMetadata().getIndices().values();
```

Comment:
```
2023-04-24T19:23:06.3310228Z   1> java.lang.ClassCastException: class java.util.Collections$UnmodifiableCollection cannot be cast to class java.util.Iterator (java.util.Collections$UnmodifiableCollection and java.util.Iterator are in module java.base of loader 'bootstrap')
2023-04-24T19:23:06.3310965Z   1> 	at org.opensearch.security.privileges.PrivilegesEvaluator$1.iterator(PrivilegesEvaluator.java:652) ~[main/:?]
```

Think you might be missing a `.iterator()` call before attempting to return, can you do this without the unchecked suppression?

You can repro the test failure with `org.opensearch.security.IndexIntegrationTests.testIndexResolveMinus` for a quick turnaround